core/src/main/java/forklift/consumer/Consumer.java

```diff
@@ -77,19 +73,20 @@
     private BlockingQueue<Runnable> blockQueue;
     private ThreadPoolExecutor threadPool;
 
-    private Callback<Consumer> outOfMessages;
+    private java.util.function.Consumer<Consumer> outOfMessages;
```

Comment:
Just an opinion: I don't necessarily agree that we should use the Java 8 `Consumer` interface  _everywhere_ it is applicable. I think it's kinda overloaded in this context and a good type name surrounding it's function is almost certainly easier to read.

But `Callback` sucks even more, and it's not really worth it to factor out all of it at this point, so yeah.